skiko/src/awtMain/kotlin/org/jetbrains/skiko/redrawer/MetalRedrawer.kt

```diff
@@ -125,13 +125,13 @@ internal class MetalRedrawer(
     override fun syncSize() = synchronized(drawLock) {
         check(isEventDispatchThread()) { "Method should be called from AWT event dispatch thread" }
         val rootPane = getRootPane(layer)
-        val globalPosition = convertPoint(layer, layer.x, layer.y, rootPane)
+        val globalPosition = convertPoint(layer.backedLayer, 0, 0, rootPane)
```

Comment:
So now we use the HardwareLayer (Canvas) origin from the Swing side, and update the CALayer to it, right? Seems about right to me

I don't even understand what the math from before should have done. It used the SkiaLayer (JPanel) origin _as coordinates whithin its own frame of reference_ which cannot be what we need. That's applying the offset of itself with respect to the content panel twice, if anything.

👍

Reply:
As I understand, this code is intended to calculate position of Skia CALayer relative to window. Seems like in examples and in our case the `layer`s origin is (0, 0) anyway, so it did work accidentally.